### ISSUE TITLE
Cannot connect to Quitter.Zone

### ISSUE THREAD

**Issue by yvolk** [andstatus/andstatus#214]
Connection to Quitter Zone doesn't work, hope to solve it with a help of its admins, see https://quitter.zone/notice/15129

1. When I try to connect to Quitter.Zone, I'm getting these error responses:

1.1. "HTTP 401 Unauthorized" response from the server (I noticed that I typed my username with wrong case: "andstatus" instead of  "AndStatus". Was this really a cause of this error?)

1.2. Typing my username as during registration this time. Getting this error: "javax.net.ssl.SSLPeerUnverifiedException: Cannot verify hostname: quitter.zone"

1.3. After I set "Allow misconfigured SSL"=on in AndStatus, I'm getting this error: "javax.net.ssl.SSLPeerUnverifiedException: No peer certificate"

Maybe Admins are changing something now...

**Comment by yvolk** [comment]
After some fixes at a server side the connection works for most but not all devices.
1. I found out that this problem occurs on _all_ Android emulators, even on Android v.4.4
2. I added another "Allow insecure SSL" hack to AndStatus (in Settings -> Logging and debugging) to make it work with Quitter.zone if it doesn't work in default (secure) mode.

**Comment by moshpirit** [comment]
Maybe it should be better to tell Holger@quitter.zone about it.

**Comment by yvolk** [comment]
@moshpirit I did this, thanks.

**Comment by moshpirit** [comment]
Great! ;) thanks for your work!

**Comment by yvolk** [comment]
Connection to Quitter.Zone fixed in AndStatus v.14.2. 
In order to connect to Quitter.Zone, you need to go to AndStatus Settings - Logging and debugging - and set the "SSL mode and security" option to "Insecure SSL".

**Comment by moshpirit** [comment]
Thanks! But this option changes all the gnu social accounts or just Quitter.zone? If all, maybe it should be better to include it when adding the account (like Mustard does)

**Comment by yvolk** [comment]
@moshpirit Currently "SSL mode and security" setting is application-wide.
But of course, ideally this "Insecure SSL" hack should be applied to concrete server (what's called "Microblogging system" in AndStatus - "GNU social instance" in this case). Unfortunately, this would mean changing the whole data model of AndStatus (and a lot of restructuring in a code...), which I didn't afford so far (for a security hack...).

**Comment by yvolk** [comment]
@holger@quitter.zone "I'm wondering why it rates quitter.zone  as unsecure SSL. In the matter of fact it is not"
AndStatus uses modern version of Apache HttpClient for Android ( https://hc.apache.org/httpcomponents-client-4.3.x/android-port.html ). I found and reused example of its usage, which works perfectly with all so far known servers, except with Quitter.Zone
The code (see it here: https://github.com/andstatus/andstatus/blob/master/app/src/org/andstatus/app/net/http/TlsSniSocketFactory.java ) raised security errors, so I searched for a solution to make it work with Quitter.Zone.
It appeared that in order to allow connection with Quitter.zone, I had to:
1. Replace "SSLCertificateSocketFactory.getDefault" with "SSLCertificateSocketFactory.getInsecure", which in Apache documentation is described as: "Returns a new instance of a socket factory with all SSL security checks disabled..."
2. Replace BrowserCompatHostnameVerifier() with AllowAllHostnameVerifier()

So connection to Quitter.Zone is currently really insecure, because most of security checks are disabled.

Everybody is welcome to find a better solution, one which doesn't require turning off security checks.
